tests/instancetype/instancetype.go

```diff
@@ -109,77 +109,90 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
-			cause := apiStatus.Status().Details.Causes[0]
-			Expect(cause.Type).To(Equal(metav1.CauseTypeFieldValueNotFound))
-			Expect(cause.Message).To(HavePrefix("Failure to find instancetype"))
-			Expect(cause.Field).To(Equal("spec.instancetype"))
-		})
-
-		It("[test_id:CNV-9089] should fail to create VM with non-existing namespaced instancetype", func() {
-			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithInstancetype("non-existing-instancetype"))
-			_, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vm, metav1.CreateOptions{})
-			Expect(err).To(HaveOccurred())
-
-			var apiStatus errors.APIStatus
-			Expect(goerrors.As(err, &apiStatus)).To(BeTrue(), "error should be type APIStatus")
-			Expect(apiStatus.Status().Details.Causes).To(HaveLen(1))
-			cause := apiStatus.Status().Details.Causes[0]
-			Expect(cause.Type).To(Equal(metav1.CauseTypeFieldValueNotFound))
-			Expect(cause.Message).To(HavePrefix("Failure to find instancetype"))
-			Expect(cause.Field).To(Equal("spec.instancetype"))
-		})
+	Context("A VirtualMachine", func() {
+		const (
+			resourceName               = "nonexistent"
+			failureCondition           = "Failure"
+			instancetypeNotFoundReason = "FailedFindInstancetype"
+			preferenceNotFoundReason   = "FailedFindPreference"
+		)
+		DescribeTable("referencing a non-existent", func(vmOptionFunc func(*virtv1.VirtualMachine), expectedReason string, createResourceFunc func(string) error) {
+			By("Creating the VM and asserting that it doesn't start and has the expected condition in place")
+			vm := libvmi.NewVirtualMachine(
+				libvmifact.NewGuestless(),
+				libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
+				vmOptionFunc,
+			)
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).Should(matcher.HaveConditionTrueWithReason(failureCondition, expectedReason))
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).Should(matcher.HavePrintableStatus(virtv1.VirtualMachineStatusStopped))
+			By("Creating the missing resource and asserting that the condition is removed and the VM starts")
+			Expect(createResourceFunc(vm.Namespace)).To(Succeed())
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).ShouldNot(matcher.HaveConditionTrueWithReason(failureCondition, expectedReason))
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).Should(matcher.BeReady())
+		},
+			Entry("[test_id:CNV-9086] cluster instance type should still be created and eventually start when missing resource created",
+				libvmi.WithClusterInstancetype(resourceName),
+				instancetypeNotFoundReason,
+				func(_ string) error {
+					instancetype := builder.NewClusterInstancetype(
+						builder.WithCPUs(1),
+						builder.WithMemory("128Mi"),
+					)
+					// FIXME(lyarwood): builder should provide WithName()
```

Comment:
Can be these comments fixed now?

Reply:
I'll hopefully address this in https://github.com/kubevirt/kubevirt/pull/14285